R/nest.R

```diff
@@ -150,21 +213,99 @@ nest.tbl_df <- function(.data, ..., .names_sep = NULL, .key = deprecated()) {
 }
 
 #' @export
-nest.grouped_df <- function(.data, ..., .names_sep = NULL, .key = deprecated()) {
+nest.grouped_df <- function(.data,
+                            ...,
+                            .by = NULL,
+                            .key = NULL,
+                            .names_sep = NULL) {
+  by <- enquo(.by)
+  if (!quo_is_null(by)) {
+    cli::cli_abort("Can't supply {.arg .by} when {.arg .data} is a grouped data frame.")
+  }
+
   if (missing(...)) {
     .key <- check_key(.key)
     cols <- setdiff(names(.data), dplyr::group_vars(.data))
     nest.tbl_df(.data, !!.key := all_of(cols), .names_sep = .names_sep)
   } else {
-    NextMethod()
+    nest.tbl_df(.data, ..., .key = .key, .names_sep = .names_sep)
```

Comment:
This was a bit unfortunate. Adding a tidy-select `.by` argument means that you can't call `NextMethod()` anymore, you have to recall the generic (or the method directly, as is done here).

I'm not sure if this will break existing code or not